sw/device/lib/testing/json/command.h

```diff
@@ -23,6 +23,7 @@ extern "C" {
     value(_, MemRead) \
     value(_, MemRead32) \
     value(_, MemWrite) \
+    value(_, PeripheralSetInstance) \
```

Comment:
As an API, this feels a little too vague for any specific test that isn't this one to use. How is a test supposed to treat this command if it needs to exercise both i2c and uart simultaneously, for example? Which one gets the use of this command to set an instance number? Are all instances always going to be interchangeable?

Mostly, I'm just thinking that this command is probably not going to be as common as its C file's name suggests. ;)

Reply:
I need to do something similar to this for the `uart_tx_rx` test, and think maybe it would be better to have the testbench `MemWrite` the instance to use, then have some kind of generic `Sync(phase_idx)` command to trigger the software to configure the correct peripheral instance.

I've needed something like a generic `Sync(phase_idx)` for other parts of a test, too.

Would that make sense to you?

The flow would be:

1. Software defines a `i2c_instance` variable.
2. Software waits for the `Sync(kPhaseCfgInstance)` message (or whatever you sync on).
3. Test bench `MemWrite`s the `i2c_instance` variable and sends the `Sync(kPhaseCfgInstance)` message.
4. Software unblocks on `Sync(...)` and configures the correct instance.

Reply:
@jwnrt That sounds good to me, yeah, especially if you were already going to have to look up symbols anyway. If the `Sync` command doesn't do anything beyond what `MemWrite` does, though, there probably isn't much need to create another command. In general, communicating through variable / memory values is more amenable to DV's backdoor mechanisms than requiring the command itself to happen.

Also, it might be nice to have an API that removes the need to look up symbol addresses in the ELF, but designing great interfaces to use across many tests is a harder problem than just using `MemWrite32`, hehe.

Reply:
I simplified the test by removing the new command and embedding the i2c_instance onto the `set_target_address` command.